### PR TITLE
SSD multi-gpu example

### PR DESCRIPTION
**Pull request by awolant**
Signed-off-by: Albert Wolant <awolant@nvidia.com>

### CODE REVIEW

File: docs/examples/pytorch/single_stage_detector/main.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    parser.add_argument('--multistep', nargs='*', type=int, default=[43, 54],
+                        help='epochs at which to decay learning rate')
+
+    # Hyperparameters
+    parser.add_argument('--learning-rate', '--lr', type=float, default=2.6e-3,
+                        help='learning rate')
+    parser.add_argument('--momentum', '-m', type=float, default=0.9,
+                        help='momentum argument for SGD optimizer')
+    parser.add_argument('--weight-decay', '--wd', type=float, default=0.0005,
+                        help='momentum argument for SGD optimizer')
```
Comment (klecki) [inline docs/examples/pytorch/single_stage_detector/main.py]: Is this supposed to be `help='momentum argument for SGD optimizer')`?

File: docs/examples/pytorch/single_stage_detector/pytorch_ssd.rst [hunk 2]
```diff
@@ -38,16 +38,14 @@ Usage
...
+  usage: main.py [-h] --data DATA [--epochs EPOCHS] [--batch-size BATCH_SIZE]
+                [--eval-batch-size EVAL_BATCH_SIZE] [--seed SEED]
+                [--evaluation [EVALUATION [EVALUATION ...]]]
+                [--multistep [MULTISTEP [MULTISTEP ...]]]
+                [--learning-rate LEARNING_RATE] [--momentum MOMENTUM]
+                [--weight-decay WEIGHT_DECAY] [--warmup WARMUP]
+                [--num-workers NUM_WORKERS] [--fp16] [--local_rank LOCAL_RANK]
+                [--data_pipeline {dali,no_dali}]
 
 All arguments with descriptions you can find in table below:
```
Comment (klecki) [inline docs/examples/pytorch/single_stage_detector/pytorch_ssd.rst]: Is this just an output of 
./main.py --help?

Comment (awolant) [reply docs/examples/pytorch/single_stage_detector/pytorch_ssd.rst]: More or less. The same thing we did for ResNet50 example, so I wanted it to be similar

File: docs/examples/pytorch/single_stage_detector/src/coco_pipeline.py [hunk 3]
```diff
@@ -12,96 +12,105 @@
...
+        except RuntimeError:
+            shard_id = 0
 
         self.input = ops.COCOReader(
-            file_root = file_root,
-            annotations_file = annotations_file,
-            ratio=True, ltrb=True,
-            random_shuffle=True)
-        self.decode = ops.nvJPEGDecoder(device="mixed", output_type=types.RGB)
+            file_root = args.train_coco_root,
```
Comment (klecki) [inline docs/examples/pytorch/single_stage_detector/src/coco_pipeline.py]: This file has very inconsistent white spaces when passing keyword arguments.

For example `file_root = args.train_coco_root,` and `skip_empty=True,`

File: docs/examples/pytorch/single_stage_detector/main.py (hunk 1 above)
Comment (awolant) [reply docs/examples/pytorch/single_stage_detector/main.py]: Fixed

**Comment by awolant** [comment]
Build 666020

**Pull request merged**